openmdao/docs/usr-guide/examples/sand_architecture.rst

```diff
@@ -0,0 +1,156 @@
+------------------------------------------------------------------------------------
+
+OpenMDAO allows you to do optimization on problems using a variety of MDAO archectures. One example of that is `Simultaneous ANalysis and Design (SAND) architecture <http://arc.aiaa.org/doi/abs/10.2514/3.9043>`_.
+
+In SAND, the optimizer minimizes the problem by varying the design variables simultaneously with the coupling variables to achieve feasibility and drive the residual constraint to zero. This means the residual needs to be expressed explicitly so we don't need any implicit components or a solver. The optimizer does it all. 
+
+Here is the code for solving the Sellar problem using the SAND architecture.
+
+.. testcode:: sand_example
+
+    from __future__ import print_function
+
+    import time
+
+    import numpy as np
+
+    from openmdao.api import Component, Group, Problem, IndepVarComp, ExecComp, NLGaussSeidel, ScipyGMRES, \
+         ScipyOptimizer, pyOptSparseDriver
+
+
+    class SellarDis1(Component):
+
+        def __init__(self):
+            super(SellarDis1, self).__init__()
+
+            self.add_param('z', val=np.zeros(2))
+            self.add_param('x', val=0.0)
+            self.add_param('y2', val=1.0)
+            self.add_param('y1', val=1.0)
+
+            self.add_output('resid1', val=1.0)
+
+        def solve_nonlinear(self, params, unknowns, resids):
+
+            z1 = params['z'][0]
+            z2 = params['z'][1]
+            x1 = params['x']
+            y2 = params['y2']
+            y1 = params['y1']
+
+            unknowns['resid1'] = z1**2 + z2 + x1 - 0.2*y2 - y1
+
+        def linearize(self, params, unknowns, resids):
+            J = {}
+
+            J['resid1','y1'] = -1.0
+            J['resid1','y2'] = -0.2
+            J['resid1','z'] = np.array([[2*params['z'][0], 1.0]])
+            J['resid1','x'] = 1.0
+
+            return J
+
+
+    class SellarDis2(Component):
+
+        def __init__(self):
+            super(SellarDis2, self).__init__()
+
+            self.add_param('z', val=np.zeros(2))
+            self.add_param('y1', val=1.0)
+            self.add_param('y2', val=1.0)
+
+            self.add_output('resid2', val=1.0)
+
+        def solve_nonlinear(self, params, unknowns, resids):
+
+            z1 = params['z'][0]
+            z2 = params['z'][1]
+            y1 = params['y1']
+            y1 = abs(y1)
+            y2 = params['y2']
+
+            unknowns['resid2'] = y1**.5 + z1 + z2 - y2
+
+        def linearize(self, params, unknowns, resids):
+            J = {}
+
+            J['resid2', 'y2'] = -1.0
+            J['resid2', 'y1'] = 0.5*params['y1']**-0.5
+            J['resid2', 'z'] = np.array([[1.0, 1.0]])
+
+            return J
+
+    class SellarSAND(Group):
+
+        def __init__(self):
+            super(SellarSAND, self).__init__()
+
+            self.add('px', IndepVarComp('x', 1.0), promotes=['*'])
```

Comment:
Hate to do this to you, but let's take out all the promote '*' and put in the variable names (since this is an example), so this one here would promote 'x' instead of '*', etc.